### PR TITLE
Passivate on nexus stop

### PR DESCRIPTION
**Pull request by adreghiciu**
Fix problem with passivation being called even if not activated.
Refactor to allow state of capability to be kept in registry.
Add more UTs.
Make IT just check a proper setup as UTs are already testing inner workings.

### CODE REVIEW

File: nexus/nexus-core-plugins/nexus-capabilities-plugin/src/main/java/org/sonatype/nexus/plugins/capabilities/api/CapabilityReference.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * http://www.gnu.org/licenses.
+ *
+ * Sonatype Nexus (TM) Open Source Version is available from Sonatype, Inc. Sonatype and Sonatype Nexus are trademarks of
+ * Sonatype, Inc. Apache Maven is a trademark of the Apache Foundation. M2Eclipse is a trademark of the Eclipse Foundation.
+ * All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.plugins.capabilities.api;
+
+/**
+ * Reference to a capability and its state.
```
Comment (bdemers) [inline nexus/nexus-core-plugins/nexus-capabilities-plugin/src/main/java/org/sonatype/nexus/plugins/capabilities/api/CapabilityReference.java:22]: Can you add more to this doc.

File: nexus/nexus-core-plugins/nexus-capabilities-plugin/src/main/java/org/sonatype/nexus/plugins/capabilities/api/CapabilityReference.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ *
+ * @since 1.10.0
```
Comment (cstamas) [inline nexus/nexus-core-plugins/nexus-capabilities-plugin/src/main/java/org/sonatype/nexus/plugins/capabilities/api/CapabilityReference.java:24]: Forgot to add `@author cstamas` ?

**Comment by cstamas** [comment]
+1

**Comment by jdillon** [comment]
+1 if this fixes the passivation on shutdown for capability which != enabled

**Pull request merged**